### PR TITLE
Misc cleanup & fixes

### PR DESCRIPTION
**Pull request by chu11**
Misc minor fixes peeled off of PR #1564

### CODE REVIEW

**Comment by chu11** [comment]
hit a #1595 , which I doubt anything I did in this PR is the cause.  Restarted.

**Comment by garlick** [comment]
This looks fine to me.  @grondo can merge if he concurs.

File: src/cmd/flux-start.c [hunk 1]
```diff
@@ -334,7 +334,7 @@ void add_args_list (char **argz, size_t *argz_len, optparse_t *opt, const char *
     optparse_getopt_iterator_reset (opt, name);
     while ((arg = optparse_getopt_next (opt, name)))
         if (argz_add  (argz, argz_len, arg) != 0)
-            log_err_exit ("subprocess_argv_append");
+            log_err_exit ("argv_add");
 }
```
Comment (grondo) [inline src/cmd/flux-start.c]: very minor: `s/argv_add/argz_add`?

**Comment by chu11** [comment]
just re-pushed with fixes, and also added 1 additional comment typo I saw about 20 mins ago.

**Comment by grondo** [comment]
Excellent! Will merge when travis reports back inevitable success!

**Comment by chu11** [comment]
hit a #731, restart builder.  And hit a hang after all tests completed successfully (no fails or errors found).  Not sure.  Maybe cleanup is slow?  Restarted builder.

```
[0;32mPASS[m: ../t/t9990-python-tests.t 44 test.request.TestRequestMethods.test_null_payload


No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
```

**Comment by chu11** [comment]
diff coverage is unsurprisingly bad, mostly b/c of log and error paths and the fact this is like ~10 lines of code change total.

**Comment by grondo** [comment]
The .007% decrease in coverage is suspicious! Merging anyway!

**Pull request merged**